src/security/ServerOptions.cc

```diff
@@ -390,7 +396,7 @@ Security::ServerOptions::loadDhParams()
     parsedDhParams.resetWithoutLocking(dhp);
 
 #else // OpenSSL 3.0+
-    const auto type = eecdhCurve.isEmpty() ? "DH" : "EC";
+    const auto type = "DH";
```

Comment:
```suggestion
    const auto type = getCurveType(eecdhCurve);
```
with a new function:

```c++
static const char *
getCurveType(const SBuf &name)
{
    if (name.isEmpty())
        return "DH"; // no curve name, default to old parser

    // see RFC 7919 Appendix A
   static const SBufList ffdhCurves = {
        SBuf("ffdhe2048"),
        SBuf("ffdhe3072"),
        SBuf("ffdhe4096"),
        SBuf("ffdhe6144"),
        SBuf("ffdhe8192")
    };
    for (const auto &ffCurve : ffdhCurves) {
        if (name == ffCurve)
            return "DH";
    }
    return "EC";
}
```

Reply:
```
tls-dh=[curve:]file
File containing DH parameters for temporary/ephemeral DH key
exchanges, optionally prefixed by a curve for ephemeral ECDH
key exchanges.
```

With the proposed code from this change request, when EC curve `name` is explicitly configured (via `tls-dh=name:file`) and does not match one of the five listed ffdhe curve names, Squid will not load DH parameters from `file` because getCurveType() returns "EC". That behavior contradicts our documentation that describes `file` as required for "File containing DH parameters for temporary/ephemeral DH key exchanges" without mentioning that it will be ignored or used for something else in the described use case.

Unlike this change request code, PR code allegedly matches our documentation for the same use case, configuring both EC curve name "for ephemeral ECDH key exchanges" and "DH parameters for temporary/ephemeral DH key exchanges".

@yadij, can we merge this PR without getCurveType() and then debate how to better support additional use cases that you want to support (and that this PR does not _break_ and, evidently, does not make more difficult to support if a simple function like getCurveType() can do it)?